src/WebUI/wwwroot/js/site.js

```diff
@@ -1,4 +1,25 @@
-﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
-// for details on configuring this project to bundle and minify static web assets.
+﻿/*
+ * Header slides
+ ----------------
+ * This code powers the "header slides", which are at the core of mobile nav
+*/
 
-// Write your JavaScript code.
+const headerSlideTriggers = document.querySelectorAll("[data-trigger-header-slide]");
+
+for (let i = 0; i < headerSlideTriggers.length; i++) {
+    headerSlideTriggers[i].addEventListener("click", function (e) {
+        const headerSlide = document.querySelector(this.getAttribute("data-trigger-header-slide"));
+
+        headerSlide.classList.toggle("is-active");
+        this.classList.toggle("is-active");
+
+        // Position header slide appropriately relative to
+        // trigger.
+        const rect = this.getBoundingClientRect();
+        hs.style.top = (rect.top + rect.height) + "px";
+        hs.style.right = (document.body.clientWidth - rect.right) + "px";
+
+        // Prevent navigation
+        e.preventDefault();
+    });
+}
```

Comment:
It's struck me that this means navigation doesn't work properly without JS enabled, which goes against the whole progressive-enhancement shtick: core functionality, including navigation, should work fully and correctly without JS enabled. It can be _enhanced_ with JS, but should work properly without. I'm not sure whether this is something that can/should be fixed here, or in Co-Design - probably the latter, I guess?